salt/modules/boto3_route53.py

```diff
@@ -717,6 +719,94 @@ def delete_hosted_zone_by_domain(Name, PrivateZone=None, region=None, key=None,
     Id = zone[0]['HostedZone']['Id']
     return delete_hosted_zone(Id=Id, region=region, key=key, keyid=keyid, profile=profile)
 
+def aws_encode(x):
+    '''
+    An implementation of the encoding required to suport AWS's domain name
+    rules defined here:
+
+        .. __: http://docs.aws.amazon.com/Route53/latest/DeveloperGuide/DomainNameFormat.html
+
+    While AWS's documentation specifies individual ASCII characters which need
+    to be encoded, we instead just try to force the string to one of
+    escaped unicode or idna depending on whether there are non-ASCII characters
+    present.
+
+    This means that we support things like ドメイン.テスト as a domain name string
+    per the idna documentation here in addition to ASCII
+
+        .. __: https://pypi.org/project/idna
+
+    This is a public method because we call it from the state in various places
+    '''
+    ret = None
+    try:
+        x.encode('ascii')
+        ret = re.sub(r'\\x([a-f0-8]{2})',
+                      _hexReplace, x.encode('unicode_escape'))
+    except UnicodeEncodeError:
+        ret = x.encode('idna')
+    except Exception as e:
+        log.error("Couldn't encode %s using either 'unicode_escape' or 'idna' codecs" % (x))
```

Comment:
You don't need to manually perform the % substitution, you can simply pass `x` as a positional argument directly after the log message. The substitution happens within our logging handler.